File: core/src/test/java/org/granchi/hollywood/ReinstateModelExceptionHandlerTest.java
package org.granchi.hollywood;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import static com.google.common.truth.Truth.assertThat;

@RunWith(MockitoJUnitRunner.class)
public class ReinstateModelExceptionHandlerTest {
    @Mock
    private Model model;

    @Mock
    private Action action;

    @Test
    public void testModelIsTheSame() throws Exception {
        Model model2 = new ReinstateModelExceptionHandler()
                .onException(model, action, new RuntimeException());

        assertThat(model).isSameAs(model2);
    }
}

File: core/src/test/kotlin/org/granchi/hollywood/ReinstateModelExceptionHandlerTest.kt
package org.granchi.hollywood

import com.google.common.truth.Truth.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class ReinstateModelExceptionHandlerTest {
    @Mock
    private lateinit var model: Model

    @Mock
    private lateinit var action: Action

    @Test
    fun testModelIsTheSame() {
        val model2 = ReinstateModelExceptionHandler()
                .onException(model, action, RuntimeException())

        assertThat(model).isSameAs(model2)
    }
}
